opus/application/static_media/js/browse.js

```diff
@@ -1475,7 +1506,8 @@ var o_browse = {
                 let lastObs = $(`${tab} .thumbnail-container`).last().data("obs");
 
                 if ((firstObs === 1 && infiniteScrollData.options.loadPrevPage === true) ||
-                    (lastObs === opus.resultCount && infiniteScrollData.options.loadPrevPage === false)) {
+                    (lastObs === opus.resultCount && infiniteScrollData.options.loadPrevPage === false) ||
+                    (lastObs === o_cart.cartCount && infiniteScrollData.options.loadPrevPage === false)) {
```

Comment:
lastObs is also depending on tab (the tab was defined in the callback function when we first init infiniteScroll). When it's in browse gallery view, it's trying to select the last rendered item in browse gallery. When it's in cart gallery view, it's trying to select the last rendered item in cart gallery.

Reply:
But you could be on the browse tab and lastObs could equal the cart count, which would be wrong. Or vice verse. Checking for resultcount or cartcount won’t work by itself. You need to check the tab name also. Also it would be good to pull the loadPrevPage out so you aren’t checking for false in each case. Would be easier to read.